p2p/pex/reactor.go

```diff
@@ -76,31 +126,108 @@ func (r *ReactorV2) OnStop() {
 	<-r.peerUpdates.Done()
 }
 
+// processPexCh implements a blocking event loop where we listen for p2p
+// Envelope messages from the pexCh.
+func (r *ReactorV2) processPexCh() {
+	defer r.pexCh.Close()
+
+	for {
+		select {
+		// outbound requests for new peers
+		case <-r.waitUntilNextRequest():
+			r.sendRequestForPeers()
```

Comment:
I've added the sending of requests in the same thread as the responding of messages. Not sure if we should be doing this in a separate go routine or now

Reply:
How long would something like this take? Ideally, we don't want to put back-pressure or block other functions of the main reactor thread.

Reply:
I imagine it's pretty quick. It just pops the next peer of the list and sends it a pex request

Reply:
Ok, i think that's fine then 👍